codalab/lib/worksheet_util.py

```diff
@@ -80,6 +83,126 @@ def worksheet_line(description, uuid):
     return '[%s]{{%s}}' % (description, uuid)
 
 
+
+############################################################
+
+
+class WorksheetResolver(object):
```

Comment:
Hm, this makes me a bit uncomfortable.  Why do the client and server have different ways to resolve uuids?  Is the problem that in the server case, we want to make another REST API call to oneself?

Another related point is that even if the implementations were different, the functionality also seems different...it'd be nice if the functionality were the same and this was just an optimization (which we can explicitly test against).  This would make it easier to reason about.

Also, can we take this opportunity to create a new Python file that contains all the resolving logic and merge that with canonicalize.py?  The path from resolving strings to uuid starting with the CLI to the DB has been one of the most convoluted parts of the system...